### PR TITLE
Honor docker-compose exit codes

### LINKED ISSUE #251: CI build reports success on failed tests
See https://github.com/StackStorm/st2-docker/actions/runs/3189178778/jobs/5202693378#step:5:20
where the BATS tests failed, but for some reason, the non-zero exit code is lost somewhere in between.

Screenshot 2022-10-05 at 11-51-28 Merge pull request #250 from mamercad_github · StackStorm_st2-docker@9609a00

This is a bug and help wanted to fix it.

### PR DESCRIPTION
**Pull request by mamercad**
Fixes #251; switch to `docker-compose run` for testing (bubbling up error codes is easier than with `docker-compose up`).

### CODE REVIEW

File: .github/workflows/st2-docker.yml [hunk 1]
```diff
@@ -32,14 +33,29 @@ jobs:
...
 
-      - name: Run st2 smoke-tests
+      - name: Sleep
         run: |
           sleep 60
-          docker-compose -f tests/st2tests.yaml up
+
+      - name: Run st2 smoke-tests
+        run: |
+          docker-compose --file tests/st2tests.yaml up --no-color 2>&1 \
```
[the remark this answers is not in the document]
Comment (armab) [reply .github/workflows/st2-docker.yml:42]: Yeah, curious how we do it in the other repos.

[the remark this answers is not in the document]
Comment (armab) [reply .github/workflows/st2-docker.yml:42]: `docker-compose run` should report the correct status, per 
https://github.com/StackStorm/st2chatops/blob/5b5fe490ee4ff510d56b4b3147133c33748d697d/.circleci/config.yml#L98-L103

That would hopefully work if you can try it.

[the remark this answers is not in the document]
Comment (mamercad) [reply .github/workflows/st2-docker.yml]: Haha, fine 😄

**Review by armab — APPROVED** [review]
Nice work!

Thanks a lot :+1:

**Comment by mamercad** [comment]
> Nice work!
> 
> Thanks a lot for the fix 👍

No problem, sorry, forgot to rebase!

**Pull request merged**